libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -40,6 +38,13 @@ export interface SimulatorProps {
    * resources
    */
   readonly factory?: ISimulatorFactory;
+
+  /**
+   * Determins if resources should be started concurrently or not. Normally we would enable this in
+   * interactive environments where determinsm is not important.
+   * @default false
+   */
+  readonly concurrency?: boolean;
```

Comment:
Now that we have dependencies, I'm not sure I see the harm in always doing concurrent starts. If the concern is that tests may fail, I would argue the tests are poorly written if they rely on the order in which resources start.

"start" in the simulator is like "deploy", and other provisioning engines deploy things concurrently when possible. They usually have a concurrently limit and maybe a way to opt out but that's more of a non-functional concern (e.g. API rate limits)

Reply:
We can make this true by default but pragmatically writing tests that are robust to concurrency is beyond the scope of this PR.

Reply:
Are you talking about unit tests in wingsdk or tests users write in wing?

Reply:
Unit tests in Wing SDK

Reply:
Fair enough, I think it's something would should do at some point but I'm cool with keeping it out of this PR